From 8aeb6cd875c13cbb2ca0e0d9900f68216878b06a Mon Sep 17 00:00:00 2001
From: xhe <xw897002528@gmail.com>
Date: Thu, 27 Dec 2018 20:40:56 +0800
Subject: [PATCH 32/48] poparser: comment may start a new block too i assume a
 block is completely parsed just before next msgid/ctxt. but the reality is
 comment may start a new block too. and if poparser dont treat comments as the
 start of new block, those comments will be parsed into as flags of old block.

---
 src/poparser.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/poparser.c b/src/poparser.c
index 7b58a1d..817bf76 100644
--- a/src/poparser.c
+++ b/src/poparser.c
@@ -132,6 +132,11 @@ enum po_error poparser_feed_line(struct po_parser *p, char* in, size_t in_len) {
 		// ignore blank lines
 		return po_success;
 	} else if (line[0] == '#') {
+		if (p->previous == po_str) {
+			if ( (t = poparser_clean(p, msg)) != po_success)
+				return t;
+		}
+
 		switch (line[1]) {
 		case ',':
 			x = &line[2];
-- 
2.25.0

